src/Features/Core/Portable/FindUsages/SourceReferenceItem.cs

```diff
@@ -41,42 +41,56 @@ internal sealed class SourceReferenceItem
         /// This entry indicates that the reference has additional value usage information which indicate
         /// it is a read/write reference, such as say 'a++'.
         /// </summary>
-        public ReferenceInfoMap ReferenceInfo { get; }
+        public ReferenceUsageInfoMap ReferenceUsageInfo { get; }
+
+        /// <summary>
+        /// Additional properties for the reference, similar to ReferenceUsageInfo.
+        /// These property values are a single string, so do not need to allocate an array of strings.
+        /// </summary>
+        public ImmutableArray<AdditionalProperty> AdditionalProperties { get; }
 
         [Obsolete]
         public SourceReferenceItem(DefinitionItem definition, DocumentSpan sourceSpan, bool isWrittenTo)
         {
             Definition = definition;
             SourceSpan = sourceSpan;
             IsWrittenTo = isWrittenTo;
-            ReferenceInfo = ReferenceInfoMap.Empty;
+            ReferenceUsageInfo = ReferenceUsageInfoMap.Empty;
         }
 
-        public SourceReferenceItem(DefinitionItem definition, DocumentSpan sourceSpan, ReferenceInfoMap referenceInfo)
+        public SourceReferenceItem(DefinitionItem definition, DocumentSpan sourceSpan, ReferenceUsageInfoMap referenceInfo)
         {
             Definition = definition;
             SourceSpan = sourceSpan;
-            ReferenceInfo = referenceInfo ?? ReferenceInfoMap.Empty;
+            ReferenceUsageInfo = referenceInfo ?? ReferenceUsageInfoMap.Empty;
         }
 
+        // Being used by TypeScript
         internal SourceReferenceItem(DefinitionItem definition, DocumentSpan sourceSpan, SymbolUsageInfo symbolUsageInfo)
-            : this(definition, sourceSpan, GetOrCreateReferenceInfo(symbolUsageInfo))
+            : this(definition, sourceSpan, GetOrCreateReferenceUsageInfo(symbolUsageInfo))
+        {
+            IsWrittenTo = symbolUsageInfo.IsWrittenTo();
```

Comment:
Do we want to initialize `AdditionalProperties` to `ImmutableArray<AdditionalProperty>.Empty` for this case? We generally don't expose immutable array properties which can have default values.